### ISSUE TITLE
Can't compile in bullseye aarch64

### ISSUE THREAD

**Issue by jofemodo** [imodularsynth/jackrtpmidid#2]
```
root@zynthian:/zynthian/zynthian-sw/jackrtpmidid# make
"make" -f nbproject/Makefile-Debug.mk QMAKE= SUBPROJECTS= .build-conf
make[1]: Entering directory '/zynthian/zynthian-sw/jackrtpmidid'
"make"  -f nbproject/Makefile-Debug.mk dist/Debug/GNU-Linux/jackrtpmidid
make[2]: Entering directory '/zynthian/zynthian-sw/jackrtpmidid'
mkdir -p build/Debug/GNU-Linux
rm -f "build/Debug/GNU-Linux/RTP_MIDI.o.d"
g++    -c -g -MMD -MP -MF "build/Debug/GNU-Linux/RTP_MIDI.o.d" -o build/Debug/GNU-Linux/RTP_MIDI.o RTP_MIDI.cpp
RTP_MIDI.cpp: In constructor ‘CRTP_MIDI::CRTP_MIDI(TMIDI_FIFO_CHAR*, unsigned char*, unsigned int*, unsigned int, TRTPDataCallback, void*)’:
RTP_MIDI.cpp:30:14: error: ‘DEFAULT_RTP_ADDRESS’ was not declared in this scope; did you mean ‘ZERO_RTP_ADDRESS’?
   30 |     RemoteIP=DEFAULT_RTP_ADDRESS;
      |              ^~~~~~~~~~~~~~~~~~~
      |              ZERO_RTP_ADDRESS
RTP_MIDI.cpp:40:30: error: no match for ‘operator=’ (operand types are ‘in6_addr’ and ‘int’)
   40 |     InvitationOnCtrlSenderIP=0;      // IP address of sender of invitation received on control port
      |                              ^
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(const in6_addr&)’
  212 | struct in6_addr
      |        ^~~~~~~~
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘const in6_addr&’
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(in6_addr&&)’
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘in6_addr&&’
RTP_MIDI.cpp:41:30: error: no match for ‘operator=’ (operand types are ‘in6_addr’ and ‘int’)
   41 |     InvitationOnDataSenderIP=0;      // IP address of sender of invitation received on data port
      |                              ^
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(const in6_addr&)’
  212 | struct in6_addr
      |        ^~~~~~~~
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘const in6_addr&’
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(in6_addr&&)’
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘in6_addr&&’
RTP_MIDI.cpp:42:22: error: no match for ‘operator=’ (operand types are ‘in6_addr’ and ‘int’)
   42 |     SessionPartnerIP=0;
      |                      ^
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(const in6_addr&)’
  212 | struct in6_addr
      |        ^~~~~~~~
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘const in6_addr&’
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(in6_addr&&)’
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘in6_addr&&’
RTP_MIDI.cpp: At global scope:
RTP_MIDI.cpp:102:5: error: no declaration matches ‘int CRTP_MIDI::InitiateSession(unsigned int, short unsigned int, short unsigned int, short unsigned int, short unsigned int, bool)’
  102 | int CRTP_MIDI::InitiateSession(unsigned int DestIP,
      |     ^~~~~~~~~
In file included from RTP_MIDI.cpp:21:
RTP_MIDI.h:175:6: note: candidate is: ‘int CRTP_MIDI::InitiateSession(in6_addr, short unsigned int, short unsigned int, short unsigned int, short unsigned int, bool)’
  175 |  int InitiateSession(struct in6_addr DestIP,
      |      ^~~~~~~~~~~~~~~
RTP_MIDI.h:151:7: note: ‘class CRTP_MIDI’ defined here
  151 | class CRTP_MIDI
      |       ^~~~~~~~~
RTP_MIDI.cpp: In member function ‘void CRTP_MIDI::RunSession()’:
RTP_MIDI.cpp:259:51: error: no match for ‘operator==’ (operand types are ‘uint32_t’ {aka ‘unsigned int’} and ‘in6_addr’)
  259 |             if ((htonl(SenderData.sin_addr.s_addr)==RemoteIP)||(RemoteIP==0))
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                       |                             |
      |                       uint32_t {aka unsigned int}   in6_addr
RTP_MIDI.cpp:259:73: error: no match for ‘operator==’ (operand types are ‘in6_addr’ and ‘int’)
  259 |             if ((htonl(SenderData.sin_addr.s_addr)==RemoteIP)||(RemoteIP==0))
      |                                                                 ~~~~~~~~^~~
      |                                                                 |         |
      |                                                                 in6_addr  int
RTP_MIDI.cpp:274:82: error: no match for ‘operator=’ (operand types are ‘in6_addr’ and ‘uint32_t’ {aka ‘unsigned int’})
  274 |                         InvitationOnCtrlSenderIP=htonl(SenderData.sin_addr.s_addr);
      |                                                                                  ^
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(const in6_addr&)’
  212 | struct in6_addr
      |        ^~~~~~~~
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘uint32_t’ {aka ‘unsigned int’} to ‘const in6_addr&’
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(in6_addr&&)’
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘uint32_t’ {aka ‘unsigned int’} to ‘in6_addr&&’
RTP_MIDI.cpp:293:51: error: no match for ‘operator==’ (operand types are ‘uint32_t’ {aka ‘unsigned int’} and ‘in6_addr’)
  293 |             if ((htonl(SenderData.sin_addr.s_addr)==RemoteIP)||(RemoteIP==0))
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
      |                       |                             |
      |                       uint32_t {aka unsigned int}   in6_addr
RTP_MIDI.cpp:293:73: error: no match for ‘operator==’ (operand types are ‘in6_addr’ and ‘int’)
  293 |             if ((htonl(SenderData.sin_addr.s_addr)==RemoteIP)||(RemoteIP==0))
      |                                                                 ~~~~~~~~^~~
      |                                                                 |         |
      |                                                                 in6_addr  int
RTP_MIDI.cpp:313:82: error: no match for ‘operator=’ (operand types are ‘in6_addr’ and ‘uint32_t’ {aka ‘unsigned int’})
  313 |                         InvitationOnDataSenderIP=htonl(SenderData.sin_addr.s_addr);
      |                                                                                  ^
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(const in6_addr&)’
  212 | struct in6_addr
      |        ^~~~~~~~
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘uint32_t’ {aka ‘unsigned int’} to ‘const in6_addr&’
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(in6_addr&&)’
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘uint32_t’ {aka ‘unsigned int’} to ‘in6_addr&&’
RTP_MIDI.cpp:418:30: error: no match for ‘operator=’ (operand types are ‘in6_addr’ and ‘int’)
  418 |             SessionPartnerIP=0;
      |                              ^
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(const in6_addr&)’
  212 | struct in6_addr
      |        ^~~~~~~~
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘const in6_addr&’
/usr/include/netinet/in.h:212:8: note: candidate: ‘constexpr in6_addr& in6_addr::operator=(in6_addr&&)’
/usr/include/netinet/in.h:212:8: note:   no known conversion for argument 1 from ‘int’ to ‘in6_addr&&’
RTP_MIDI.cpp:447:43: error: cannot convert ‘in6_addr’ to ‘uint32_t’ {aka ‘unsigned int’}
  447 |             AdrEmit.sin_addr.s_addr=htonl(SessionPartnerIP);    // V0.6 : use SessionPartnerIP, not Remote IP
      |                                           ^~~~~~~~~~~~~~~~
      |                                           |
      |                                           in6_addr
In file included from RTP_netdriver.h:21,
                 from RTP_MIDI.h:19,
                 from RTP_MIDI.cpp:21:
/usr/include/netinet/in.h:378:33: note:   initializing argument 1 of ‘uint32_t htonl(uint32_t)’
  378 | extern uint32_t htonl (uint32_t __hostlong)
      |                        ~~~~~~~~~^~~~~~~~~~
make[2]: *** [nbproject/Makefile-Debug.mk:72: build/Debug/GNU-Linux/RTP_MIDI.o] Error 1
make[2]: Leaving directory '/zynthian/zynthian-sw/jackrtpmidid'
make[1]: *** [nbproject/Makefile-Debug.mk:63: .build-conf] Error 2
make[1]: Leaving directory '/zynthian/zynthian-sw/jackrtpmidid'
make: *** [nbproject/Makefile-impl.mk:40: .build-impl] Error 2

```

**Comment by bbouchez** [comment]
Hi José,
it seems that the problem is not related to 64-bits platform itself, but from the IPv6 code which was added by jwoillez.

My original code was for IPv4 only and the addresses are coded using uint32_t type. As far as I can see, the problem is that the compiler complains about the fact that all default IP addresses written as 32 bits are not declared in IPv6 dedicated types (in6_addr type). Since in6_addr variables can not be initialized from a 32-bit value, the compiler does not know what to do.

**Comment by bbouchez** [comment]
Issue has been solved in V0.7 after removing merged IPv6 code (IPv6 is almost never used in RTP-MIDI configuration)
Issue is closed

**Comment by bbouchez** [comment]
Issue closed
